packages/devtools/lib/src/utils.dart

```diff
@@ -152,31 +152,18 @@ String longestFittingSubstring(
 Uri getTrimmedUri(String value) {
   final startingUri = Uri.parse(value);
 
-  // Path should be of the form "/72K34Xmq0X0=/", but it could have trailing
-  // characters that we should trim.
-  final path = startingUri.path;
-
-  // Find the end of token index (index of the second slash).
-  int endOfTokenIndex = -1;
-  int slashCount = 0;
-  for (int i = 0; i < path.length; i++) {
-    if (path[i] == '/') {
-      slashCount++;
-    }
-    if (slashCount == 2) {
-      endOfTokenIndex = i;
-      break;
-    }
-  }
-
-  if (endOfTokenIndex == -1) {
+  // [startingUri.path] should be of the form "/72K34Xmq0X0=/", but it could
+  // have trailing characters. Trim any excess chars beyond the second slash in
+  // the uri path.
+  final pathParts = startingUri.path.split('/');
+  if (pathParts.length < 2) {
```

Comment:
This special case isn't actually needed.
Take takes at most 2 elements, not exactly 2 elements.
https://api.dartlang.org/stable/1.10.1/dart-core/List/take.html

Reply:
Correction. It is because of the added trailing slash. disregard my previous comment.

Reply:
I am still going to need to return early here because if we get a uri with no path or with just a trailing slash, then we shouldn't continue to the `indexOf(newPath)` and substring parts of this method. So this check is more or a short circuit check rather than a NPE avoidance check for the `take` method.

Rather than checking `if (newPath == '' || newPath == '/')` to return early, I'll leave this here as I think it is cleaner and more robust.